tests/test_box.py

```diff
@@ -401,3 +407,43 @@ def fn(a, b):
         fn(1)
 
     excinfo.match('b')
+
+
+def test_chainbox_put_changes_box():
+    testbox = picobox.Box()
+    testchainbox = picobox.ChainBox(testbox)
+
+    with pytest.raises(KeyError):
+        testchainbox.get('the-key')
+    testchainbox.put('the-key', 42)
+
+    assert testbox.get('the-key', 42)
+
+
+def test_chainbox_get_chained():
+    testbox_a = picobox.Box()
+    testbox_a.put('the-key', 42)
+
+    testbox_b = picobox.Box()
+    testbox_b.put('the-key', 13)
+    testbox_b.put('the-pin', 12)
+
+    testchainbox = picobox.ChainBox(testbox_a, testbox_b)
+
+    assert testchainbox.get('the-key') == 42
+    assert testchainbox.get('the-pin') == 12
+
+
+def test_chainbox_isinstance_box():
+    assert isinstance(picobox.ChainBox(), picobox.Box)
+
+
+@pytest.mark.parametrize('name', [
+    name
+    for name, _ in inspect.getmembers(picobox.Box) if not name.startswith('_')
```

Comment:
nit: perhaps you could add a predicate that would only return "methods" (or "functions" in case of Python 3) of the class

Reply:
I would need to have a list comprehension anyway to select only `name` part, so I decided to go with just list comprehensions for now.